nion/swift/Panel.py

```diff
@@ -376,6 +395,108 @@ def context_menu_event(self, x: int, y: int, gx: int, gy: int) -> bool:
+                drawing_context.fill_text(self.title, canvas_size.width // 2, canvas_size.height - self.__text_offset)
+
+            font_string = self.__font
+            if self.__get_font_metrics:
+                font_metrics = self.__get_font_metrics(font_string, self.title)
+                title_rect = Geometry.IntRect.from_tlhw(
+                    canvas_size.height - self.__text_offset - (font_metrics.height // 2),
+                    (canvas_size.width // 2) - (font_metrics.width // 2),
+                    font_metrics.height,
+                    font_metrics.width)
+            else:
+                 # Without knowing how big the text is, need to do a clunky fallback
+                 # Close button and pencil is roughly 40 pixels wide
+                title_rect = Geometry.IntRect.from_tlhw(0, 40, canvas_size.height, canvas_size.width - 80)
+        return title_rect
+
+
+    def _draw_edit_button(self, drawing_context: DrawingContext.DrawingContext, title_rect: typing.Optional[Geometry.IntRect]) -> typing.Optional[Geometry.IntRect]:
+        control_title_margin = 3
+        control_margin = 2
+        control_width = 10
+        canvas_size = self.canvas_size
+        control_rect: typing.Optional[Geometry.IntRect] = None
+        if canvas_size and title_rect:
+            with drawing_context.saver():
+                drawing_context.begin_path()
+                close_box_left = title_rect.right + control_margin + control_title_margin # canvas_size.width - (40 - 7)
+                close_box_right = close_box_left + control_width # canvas_size.width - (40 - 13)
+                close_box_top = canvas_size.height // 2 - 5
+                close_box_bottom = canvas_size.height // 2 + 5
+                drawing_context.move_to(close_box_left, close_box_bottom)
+                drawing_context.line_to(close_box_left + 4, close_box_bottom - 1)
+                drawing_context.line_to(close_box_left + 10, close_box_bottom - 8)
+                drawing_context.line_to(close_box_left + 8, close_box_bottom - 10)
+                drawing_context.line_to(close_box_left + 1, close_box_bottom - 4)
+                drawing_context.line_to(close_box_left, close_box_bottom)
+                drawing_context.move_to(close_box_left + 4, close_box_bottom - 1)
+                drawing_context.line_to(close_box_left, close_box_bottom - 3)
+                drawing_context.line_width = 1
+                drawing_context.line_cap = "round"
+                drawing_context.stroke_style = self.__control_style
+                drawing_context.stroke()
+            control_rect = Geometry.IntRect.from_tlhw(close_box_top - control_margin, close_box_left - control_margin, control_width + 2 * control_margin, control_width + 2 * control_margin)
+            self.__edit_control_rect = control_rect
+        return control_rect
     def _repaint(self, drawing_context: DrawingContext.DrawingContext) -> None:
```

Comment:
Should be a single blank line here. Please consider using `flake8 --ignore=E501 nion/swift/Panel.py nion/swift/DisplayPanel.py` and addressing any issues specific to this PR. You may need to install `conda install flake8` in your environment.